src/Analysis/Ast/Impl/Diagnostics/ErrorCodes.cs

```diff
@@ -29,5 +29,6 @@ public static class ErrorCodes {
         public const string ReturnInInit = "return-in-init";
         public const string TypingNewTypeArguments = "typing-newtype-arguments";
         public const string TypingGenericArguments = "typing-generic-arguments";
+        public const string WrongArgumentCount = "wrong-argument-count";
```

Comment:
On second look, how is this differing from the other two already defined messages, `too-many-function-arguments` and `parameter-missing`?

Reply:
You have a good point. Those two messages are used when building argument sets. What I could do instead of explicitly checking for the number of arguments is once again do the whole 

```
o.setParameters(...)
```

And I believe that if there were not enough arguments we would get them ourselves. What I do not like about those error messages is that they do not say how many arguments needed but I might be able to add that functionality.

Reply:
After looking into more, those errorcodes are used in ArgumentSet DiagnosticEntries, but those entries get put in the Errors list in ArgumentSet, which is not actually shown to the user. 

For that reason I think it's better to make something separate. 

For example:

```
def hello(a, b):
  pass

hello(1,2,1)
```

Will not show an error to the user. We record it internally.

However, for something like:

```
NewType(1)
```
We know for sure this is an error so we can report this.

We have two options 

* Make the Errors field in ArgumentSet accessible to the user (I believe it was not because of low confidence in the argument parsing) and we can use the same errors codes in NewType.

* Make a new error code for functions that we know the definitions of and we are confident in error checking.

Also we should consider what the Errors field in ArgumentSet is even used for. At the moment I don't see where the diagnostic entries in Errors are actually used, only their count.